android/src/main/java/com/wix/pagedcontacts/contacts/ContactsProvider.java

```diff
@@ -140,4 +154,232 @@ private Cursor queryContacts(QueryParams params) {
+            ops.add(op.build());
+        }
+
+        for (int i = 0; i < numOfUrls; i++) {
+            op = ContentProviderOperation.newInsert(ContactsContract.Data.CONTENT_URI)
+                    .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID, 0)
+                    .withValue(ContactsContract.Data.MIMETYPE, CommonDataKinds.Website.CONTENT_ITEM_TYPE)
+                    .withValue(CommonDataKinds.Website.URL, urls[i]);
+            ops.add(op.build());
+        }
+
+        for (int i = 0; i < numOfEmails; i++) {
+            op = ContentProviderOperation.newInsert(ContactsContract.Data.CONTENT_URI)
+                    .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID, 0)
+                    .withValue(ContactsContract.Data.MIMETYPE, CommonDataKinds.Email.CONTENT_ITEM_TYPE)
+                    .withValue(CommonDataKinds.Email.ADDRESS, emails[i])
+                    .withValue(CommonDataKinds.Email.TYPE, emailsTypes[i])
+                    .withValue(CommonDataKinds.Email.LABEL, emailsLabels[i]);
+            ops.add(op.build());
+        }
+
+        ReadableArray postalAddresses = contact.hasKey("postalAddresses") ? contact.getArray("postalAddresses") : null;
+        if (postalAddresses != null) {
+            for (int i = 0; i < postalAddresses.size(); i++) {
+                ReadableMap address = postalAddresses.getMap(i);
+
+                op = ContentProviderOperation.newInsert(ContactsContract.Data.CONTENT_URI)
+                        .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID, 0)
+                        .withValue(ContactsContract.Data.MIMETYPE, CommonDataKinds.StructuredPostal.CONTENT_ITEM_TYPE)
+                        .withValue(CommonDataKinds.StructuredPostal.TYPE, mapStringToPostalAddressType(address.getString("label")))
+                        .withValue(CommonDataKinds.StructuredPostal.LABEL, address.getString("label"))
+                        .withValue(CommonDataKinds.StructuredPostal.STREET, address.getString("street"))
+                        .withValue(CommonDataKinds.StructuredPostal.CITY, address.getString("city"))
+                        .withValue(CommonDataKinds.StructuredPostal.REGION, address.getString("state"))
+                        .withValue(CommonDataKinds.StructuredPostal.POSTCODE, address.getString("postCode"))
+                        .withValue(CommonDataKinds.StructuredPostal.COUNTRY, address.getString("country"));
+
+                ops.add(op.build());
+            }
+        }
+
+            try {
+            ContentResolver cr = this.context.getContentResolver();
+            ContentProviderResult[] result = cr.applyBatch(ContactsContract.AUTHORITY, ops);
+
+            promise.resolve(null);
+
+            } catch (Exception e) {
+                promise.reject(e);
+            }
+//        }
+    }
+
+    private int mapStringToPhoneType(String label) {
+        int phoneType;
```

Comment:
They are similar to the contacts returned by `react-native-paged-contacts` (Same fields mappings, but the other way around) do we really want to expose it separately?

only difference I made is `imageUrl` because I explicitly use a URL for the contact's image.